docs/high_availability_installation_guide.md

```diff
@@ -0,0 +1,284 @@
+
+**Clair DB cluster**: The PostgreSQL is used by Clair to store the vulnerability data which will be used by scanning the images. User should follow it's best practice to make it HA protected 
+
+**Shared Storage**: The shared storage is used for storing Docker Volumes used by Harbor. Images pushed by users are actually stored in this shared storage. The shared storage makes sure that multiple Harbor instances have consistent storage backend. Shared Storages can be Swift, NFS, S3, azure, GCS Ceph or OSS. User should follow its best practice to make it HA protect. 
+
+**Redis**: The purpose of having Redis is to store UI session data and store the registry metadata cache. When one Harbor instance fails or the load balancer routes a user request to another Harbor instance, any Harbor instance can query the Redis to retrieve session information to make sure the end-user has a continued session. User should follow the best practice of Redis to make it HA protect.  
+
+From the above high availability architecture, we can see that we don't setup LB per stateless services. Instead we group those stateless service as a group. The communicate between each services are protected by host based docker network with isolation. **Please Note!** As the component communicate with each other through rest API. You can always define the group granularity according to your use scenarios. 
+
+### Limitation 
+
+Currently Harbor doesn’t support and Notary in HA scenario. Which means it don't support Content trust for this HA setup. 
+
+## Setup 
+
+Follow the setup instruction in this section we build a Harbor high availability deployment as the follow figure shows. You can setup more Harbor nodes if needed. 
+
+![setup](img/ha/LabInstallation.png)
+
+### Prerequisites 
+
+ - 1> MariaDB cluster(Harbor-DB,192.168.1.215, currently harbor use mariadb 10.2.10) 
+ - 2> Shared Storage (Swift Server. 192.168.1.216. **Important!** The shared storage is replaceable you can choose other shared storage, just need to make sure the storage you used is supported by registry https://docs.docker.com/registry/storage-drivers) 
+ - 3> Redis cluster(192.168.1.217)
+ - 4> PostgreSQL (Clair DB 192.168.1.50. Note! The PostgreSQL is optional, it only needed when you use the vulnerability scanning function, currently harbor use PostgreSQL 9.6.5) 
+
+**Important!** Item 1,2,3,4 are considered external components to Harbor. Before configuring Harbor HA, we assume these components are present and all of them are HA protected. Otherwise, any of these components can be a single point of failure. 
+
+**Tips!**
+
+If you are only setting up the HA for POC purpose. You can use docker to run MariaDB, Redis and PostgreSQL in one OS with follow command. 
+```
+#> docker run --name redis-server -p 6379:6379 -d redis 
```

Comment:
suggestion: "#>" is not needed for each command. We can tell the user to run the script with root user or sudo user if 'root' is required.